### ISSUE TITLE
[debops.netbox ] Netbox and remote Postgresql server

### ISSUE THREAD

**Issue by SoulRaven** [debops/debops#1327]
possibile i don't understand how the dependent roles are working
but if i define for example: netbox__postgresql__dependent_roles inside the netbox.yml
normally the roles will be set on the remote postgresql server, if so, way the postgresql server is installed on netbox host and the role try to config netbox instance of postgresql?

I am missing something?

**Comment by drybjed** [comment]
The system is working like this: you can install PostgreSQL database on a `db` server using the `postgresql_server` role. Then, on one or multiple `appX` servers, you can apply the `postgresql` role, with the `postgresql__server` variable pointing to the `db` server. Any roles that integrate with the `postgresql` role (like `netbox`), will automatically create the databases and PG roles on the remote PostgreSQL server.

You don't need to do anything in the `netbox__postgresql__dependent_roles` variable, this will happen automatically as long as you correctly configure the `postgresql__server` variable (notice the double-underscore, which means that this variable isued by the `postgresql` role, not `postgresql_server` role).

If you have any more questions, let me know.

**Comment by SoulRaven** [comment]
so, if i configure the postgresql__server inside the netbox host_vars and point to the postgresql server, no local postgresql server will be installed locally, and all the configs will be make on remote postgresql server? right?

**Comment by drybjed** [comment]
Yes, that's correct. The `postgresql` role will install the PosgreSQL "client" package on the host that you apply it to, along with the Python and other libraries, but will not configure the server on it. This also applies if you want to install the Netbox application on the host with PostgreSQL server - you need to run the `postgresql_server` playbook on that host before running the `netbox` playbook.
